src/Microsoft.AspNetCore.ResponseCaching/ResponseCachingContext.cs

```diff
@@ -118,15 +144,37 @@ internal bool OnResponseStarting()
 
         internal void FinalizeCaching()
         {
-            if (CacheResponse)
+            // Don't cache errors? 404 etc
+            if (CacheResponse && HttpContext.Response.StatusCode == 200)
             {
                 // Store the buffer to cache
-                var cacheEntry = new ResponseCachingEntry();
-                cacheEntry.StatusCode = HttpContext.Response.StatusCode;
-                foreach (var pair in HttpContext.Response.Headers)
+                var cacheEntry = new ResponseCachingEntry()
+                {
+                    Created = DateTimeOffset.UtcNow,
```

Comment:
This should be set to the `Date` header from the response? That header contains the information on when the response was actually generated.